### ISSUE TITLE
Filter Flows by most recent flow run state

### ISSUE THREAD

**Issue by billpalombi** [PrefectHQ/prefect#9405]
### First check

- [X] I added a descriptive title to this issue.
- [X] I used the GitHub search to find a similar request and didn't find it.
- [X] I searched the Prefect documentation for this feature.

### Prefect Version

2.x

### Describe the current behavior

Currently, I cannot make a single request to `/flows/filter` and get back all of the flows for which the run that was scheduled to start most recently is in a certain state.

### Describe the proposed behavior

I can make a single request to `/flows/filter` and get back all of the flows for which the run that was scheduled to start most recently is in a certain state.

I don't want to filter by the state of the most recent completed run. I want to filter by the state of the run with the most recent start time.

### Example Use

Most commonly, I want to see the flows for which the most recent run failed, but I also commonly want to see runs that are running or late. 

### Additional context

This enhancement will support the flow and deployments page redesign.

**Comment by serinamarie** [comment]
@billpalombi Is this still a high priority issue given that this enhancement this was meant to support is released?

**Comment by billpalombi** [comment]
Yes, this is still high priority, it would be great to have this.
